In [258]:
#Import dependencies
import pandas as pd

In [259]:
#Read CSV Files
school_data = pd.read_csv('Resources/schools_complete.csv')
student_data = pd.read_csv('Resources/students_complete.csv')

In [260]:
#Combine the data into a single dataset
school_data_complete = pd.merge(student_data, school_data, how="left", on=["school_name", "school_name"])
school_data_complete.head()

Student ID       student_name gender grade        school_name  \
0           0       Paul Bradley      M   9th  Huang High School   
1           1       Victor Smith      M  12th  Huang High School   
2           2    Kevin Rodriguez      M  12th  Huang High School   
3           3  Dr. Richard Scott      M  12th  Huang High School   
4           4         Bonnie Ray      F   9th  Huang High School   

   reading_score  math_score  School ID      type  size   budget  
0             66          79          0  District  2917  1910635  
1             94          61          0  District  2917  1910635  
2             90          60          0  District  2917  1910635  
3             67          58          0  District  2917  1910635  
4             97          84          0  District  2917  1910635

# District Summary

In [261]:
# Calculate the total number of unique schools
unique_school_name = school_data_complete['school_name'].unique()

total_school_number=len(unique_school_name)
total_school_number


15

In [262]:
# Calculate the total number of students
total_students_number = school_data_complete["student_name"].count()
total_students_number

39170

In [263]:
#Calculate the total budget
total_budget = school_data["budget"].sum()
total_budget

 


24649428

In [264]:
#Calculate the average (mean) math score
average_math_score = school_data_complete["math_score"].mean()
average_math_score

78.98537145774827

In [265]:
#Calculate the average (mean) reading score
average_reading_score = school_data_complete["reading_score"].mean()
average_reading_score

81.87784018381414

In [266]:
#Use the code provided to calculate the percentage of students who passed math 
students_passing_math = school_data_complete.loc[school_data_complete["math_score"] >= 70]
number_students_passing_math = students_passing_math["Student ID"].count()

percent_passing_math = (number_students_passing_math / total_students_number) * 100

percent_passing_math

74.9808526933878

In [267]:
#Calculate the percentage of students who passed reading
students_passing_reading = school_data_complete.loc[school_data_complete["reading_score"] >= 70]
number_students_passing_reading = students_passing_reading["Student ID"].count()

percent_passing_reading = (number_students_passing_reading / total_students_number) * 100
percent_passing_reading

85.80546336482001

In [268]:
#Use the code provided to calculate the percentage of students that passed both math and reading 
Overall_passing = school_data_complete[(school_data_complete['math_score'] >= 70) & (school_data_complete['reading_score'] >= 70)]['Student ID'].count()/total_students_number*100
Overall_passing

65.17232575950983

In [269]:
#Create a new DataFrame for the above calculations called district_summary
district_summary = pd.DataFrame({
    "Total Schools": total_school_number,
    "Total Students": f"{total_students_number:,}",
    "Total Budget": f"${total_budget:,.2f}",
    "Average Math Score": f"{average_math_score:.6f}",
    "Average Reading Score": f"{average_reading_score:.5f}",
    "% Passing Math": f"{percent_passing_math:.6f}",
    "% Passing Reading": f"{percent_passing_reading:.6f}",
    "% Overall Passing": f"{Overall_passing: .6f}"
}, index=[0])

district_summary

Total Schools Total Students    Total Budget Average Math Score  \
0             15         39,170  $24,649,428.00          78.985371   

  Average Reading Score % Passing Math % Passing Reading % Overall Passing  
0              81.87784      74.980853         85.805463         65.172326

# School Summary


In [270]:
# Use the code provided to select the school type
school_type = school_data.set_index('school_name')['type']



In [271]:
# Group by school name
school_name = school_data_complete.set_index('school_name').groupby(['school_name'])

In [272]:
# Calculate the total student count 
total_student = school_name['Student ID'].count()
total_student

school_name
Bailey High School       4976
Cabrera High School      1858
Figueroa High School     2949
Ford High School         2739
Griffin High School      1468
Hernandez High School    4635
Holden High School        427
Huang High School        2917
Johnson High School      4761
Pena High School          962
Rodriguez High School    3999
Shelton High School      1761
Thomas High School       1635
Wilson High School       2283
Wright High School       1800
Name: Student ID, dtype: int64

In [273]:
# Use the code provided to calculate the per capita spending 
per_school_budget = school_data.groupby("school_name")["budget"].sum()
per_school_capita = per_school_budget/total_student
per_school_budget
per_school_capita


school_name
Bailey High School       628.0
Cabrera High School      582.0
Figueroa High School     639.0
Ford High School         644.0
Griffin High School      625.0
Hernandez High School    652.0
Holden High School       581.0
Huang High School        655.0
Johnson High School      650.0
Pena High School         609.0
Rodriguez High School    637.0
Shelton High School      600.0
Thomas High School       638.0
Wilson High School       578.0
Wright High School       583.0
dtype: float64

In [274]:
# Calculate the average test scores
per_school_math = student_data.groupby("school_name") ["math_score"].mean()
per_school_reading = student_data.groupby("school_name") ["reading_score"].mean()
per_school_math
per_school_reading


school_name
Bailey High School       81.033963
Cabrera High School      83.975780
Figueroa High School     81.158020
Ford High School         80.746258
Griffin High School      83.816757
Hernandez High School    80.934412
Holden High School       83.814988
Huang High School        81.182722
Johnson High School      80.966394
Pena High School         84.044699
Rodriguez High School    80.744686
Shelton High School      83.725724
Thomas High School       83.848930
Wilson High School       83.989488
Wright High School       83.955000
Name: reading_score, dtype: float64

In [275]:
#Calculate the number of schools with math scores of 70 or higher 
students_passing_math = student_data[student_data["math_score"]>=70]
school_students_passing_math = students_passing_math.groupby("school_name")["Student ID"].count()
students_passing_math
school_students_passing_math

school_name
Bailey High School       3318
Cabrera High School      1749
Figueroa High School     1946
Ford High School         1871
Griffin High School      1371
Hernandez High School    3094
Holden High School        395
Huang High School        1916
Johnson High School      3145
Pena High School          910
Rodriguez High School    2654
Shelton High School      1653
Thomas High School       1525
Wilson High School       2143
Wright High School       1680
Name: Student ID, dtype: int64

In [276]:
# Calculate the number of schools with reading scores of 70 or higher
students_passing_reading = student_data[student_data["reading_score"]>=70]
school_students_passing_reading = students_passing_reading.groupby("school_name")["Student ID"].count()
students_passing_reading
school_students_passing_reading



school_name
Bailey High School       4077
Cabrera High School      1803
Figueroa High School     2381
Ford High School         2172
Griffin High School      1426
Hernandez High School    3748
Holden High School        411
Huang High School        2372
Johnson High School      3867
Pena High School          923
Rodriguez High School    3208
Shelton High School      1688
Thomas High School       1591
Wilson High School       2204
Wright High School       1739
Name: Student ID, dtype: int64

In [277]:
#Use the provided code to calculate the schools that passed both math and reading with scores of 70 or higher (2 points)
overall_pass = school_data_complete[(school_data_complete['reading_score'] >= 70) & (school_data_complete['math_score'] >= 70)]
overall_pass

Student ID     student_name gender grade         school_name  \
4               4       Bonnie Ray      F   9th   Huang High School   
5               5    Bryan Miranda      M   9th   Huang High School   
6               6    Sheena Carter      F  11th   Huang High School   
8               8     Michael Roth      M  10th   Huang High School   
9               9   Matthew Greene      M  10th   Huang High School   
...           ...              ...    ...   ...                 ...   
39165       39165     Donna Howard      F  12th  Thomas High School   
39166       39166        Dawn Bell      F  10th  Thomas High School   
39167       39167   Rebecca Tanner      F   9th  Thomas High School   
39168       39168     Desiree Kidd      F  10th  Thomas High School   
39169       39169  Carolyn Jackson      F  11th  Thomas High School   

       reading_score  math_score  School ID      type  size   budget  
4                 97          84          0  District  2917  1910635  
5                 94          94          0  District  2917  1910635  
6                 82          80          0  District  2917  1910635  
8                 95          87          0  District  2917  1910635  
9                 96          84          0  District  2917  1910635  
...              ...         ...        ...       ...   ...      ...  
39165             99          90         14   Charter  1635  1043130  
39166             95          70         14   Charter  1635  1043130  
39167             73          84         14   Charter  1635  1043130  
39168             99          90         14   Charter  1635  1043130  
39169             95          75         14   Charter  1635  1043130  

[25528 rows x 11 columns]

In [278]:
#Use the provided code to calculate the passing rates
per_school_passing_math = school_students_passing_math/total_student * 100
per_school_passing_reading = school_students_passing_reading/total_student * 100



In [279]:
# Use the provided code to calculate the passing rates
overall_passing_rate = (per_school_passing_math + per_school_passing_reading) /2
overall_passing_rate

school_name
Bailey High School       74.306672
Cabrera High School      95.586652
Figueroa High School     73.363852
Ford High School         73.804308
Griffin High School      95.265668
Hernandez High School    73.807983
Holden High School       94.379391
Huang High School        73.500171
Johnson High School      73.639992
Pena High School         95.270270
Rodriguez High School    73.293323
Shelton High School      94.860875
Thomas High School       95.290520
Wilson High School       95.203679
Wright High School       94.972222
Name: Student ID, dtype: float64

In [280]:
# Create a new DataFrame for the above calculations called per_school_summary
per_school_summary = pd.DataFrame({
    "School Type": school_type,
    "Total Students": total_student,
    "Total School Budget": per_school_budget,
    "Per Student Budget": per_school_capita,
    "Average Reading Score": per_school_reading,
    "Average_Math_Score": per_school_math,
    "Passing Math Percent": per_school_passing_math,
    "Passing Reading Percent": per_school_passing_reading,
    "Passing Rate": overall_passing_rate

})

# Formatting
per_school_summary ["Total School Budget"] = per_school_summary["Total School Budget"].map('${:,.2f}'.format)
per_school_summary ["Per Student Budget"] = per_school_summary["Per Student Budget"].map("${:,.2f}".format)

#Display the DataFrame
per_school_summary

School Type  Total Students Total School Budget  \
school_name                                                             
Bailey High School       District            4976       $3,124,928.00   
Cabrera High School       Charter            1858       $1,081,356.00   
Figueroa High School     District            2949       $1,884,411.00   
Ford High School         District            2739       $1,763,916.00   
Griffin High School       Charter            1468         $917,500.00   
Hernandez High School    District            4635       $3,022,020.00   
Holden High School        Charter             427         $248,087.00   
Huang High School        District            2917       $1,910,635.00   
Johnson High School      District            4761       $3,094,650.00   
Pena High School          Charter             962         $585,858.00   
Rodriguez High School    District            3999       $2,547,363.00   
Shelton High School       Charter            1761       $1,056,600.00   
Thomas High School        Charter            1635       $1,043,130.00   
Wilson High School        Charter            2283       $1,319,574.00   
Wright High School        Charter            1800       $1,049,400.00   

                      Per Student Budget  Average Reading Score  \
school_name                                                       
Bailey High School               $628.00              81.033963   
Cabrera High School              $582.00              83.975780   
Figueroa High School             $639.00              81.158020   
Ford High School                 $644.00              80.746258   
Griffin High School              $625.00              83.816757   
Hernandez High School            $652.00              80.934412   
Holden High School               $581.00              83.814988   
Huang High School                $655.00              81.182722   
Johnson High School              $650.00              80.966394   
Pena High School                 $609.00              84.044699   
Rodriguez High School            $637.00              80.744686   
Shelton High School              $600.00              83.725724   
Thomas High School               $638.00              83.848930   
Wilson High School               $578.00              83.989488   
Wright High School               $583.00              83.955000   

                       Average_Math_Score  Passing Math Percent  \
school_name                                                       
Bailey High School              77.048432             66.680064   
Cabrera High School             83.061895             94.133477   
Figueroa High School            76.711767             65.988471   
Ford High School                77.102592             68.309602   
Griffin High School             83.351499             93.392371   
Hernandez High School           77.289752             66.752967   
Holden High School              83.803279             92.505855   
Huang High School               76.629414             65.683922   
Johnson High School             77.072464             66.057551   
Pena High School                83.839917             94.594595   
Rodriguez High School           76.842711             66.366592   
Shelton High School             83.359455             93.867121   
Thomas High School              83.418349             93.272171   
Wilson High School              83.274201             93.867718   
Wright High School              83.682222             93.333333   

                       Passing Reading Percent  Passing Rate  
school_name                                                   
Bailey High School                   81.933280     74.306672  
Cabrera High School                  97.039828     95.586652  
Figueroa High School                 80.739234     73.363852  
Ford High School                     79.299014     73.804308  
Griffin High School                  97.138965     95.265668  
Hernandez High School                80.862999     73.80798

# Highest-Performing Schools (by % Overall Passing)

In [281]:
# Sort the highest performing schools by passing rate in descending order
top_perform = per_school_summary.sort_values("Passing Rate", ascending=True)
top_perform_formatted = top_perform.head().style.format({
    'Total Students': '{:}',
    'Total School Budget': "${:.2f}",
    'Passing Math Percent': lambda x: "{:.6f}".format(x) if isinstance(x, (int, float)) else x,
    'Passing Reading Percent': lambda x: "{:.6f}".format(x) if isinstance(x, (int, float)) else x,
    'Passing Rate': lambda x: "{:.6f}".format(x) if isinstance(x, (int, float)) else x
})

top_perform






School Type  Total Students Total School Budget  \
school_name                                                             
Rodriguez High School    District            3999       $2,547,363.00   
Figueroa High School     District            2949       $1,884,411.00   
Huang High School        District            2917       $1,910,635.00   
Johnson High School      District            4761       $3,094,650.00   
Ford High School         District            2739       $1,763,916.00   
Hernandez High School    District            4635       $3,022,020.00   
Bailey High School       District            4976       $3,124,928.00   
Holden High School        Charter             427         $248,087.00   
Shelton High School       Charter            1761       $1,056,600.00   
Wright High School        Charter            1800       $1,049,400.00   
Wilson High School        Charter            2283       $1,319,574.00   
Griffin High School       Charter            1468         $917,500.00   
Pena High School          Charter             962         $585,858.00   
Thomas High School        Charter            1635       $1,043,130.00   
Cabrera High School       Charter            1858       $1,081,356.00   

                      Per Student Budget  Average Reading Score  \
school_name                                                       
Rodriguez High School            $637.00              80.744686   
Figueroa High School             $639.00              81.158020   
Huang High School                $655.00              81.182722   
Johnson High School              $650.00              80.966394   
Ford High School                 $644.00              80.746258   
Hernandez High School            $652.00              80.934412   
Bailey High School               $628.00              81.033963   
Holden High School               $581.00              83.814988   
Shelton High School              $600.00              83.725724   
Wright High School               $583.00              83.955000   
Wilson High School               $578.00              83.989488   
Griffin High School              $625.00              83.816757   
Pena High School                 $609.00              84.044699   
Thomas High School               $638.00              83.848930   
Cabrera High School              $582.00              83.975780   

                       Average_Math_Score  Passing Math Percent  \
school_name                                                       
Rodriguez High School           76.842711             66.366592   
Figueroa High School            76.711767             65.988471   
Huang High School               76.629414             65.683922   
Johnson High School             77.072464             66.057551   
Ford High School                77.102592             68.309602   
Hernandez High School           77.289752             66.752967   
Bailey High School              77.048432             66.680064   
Holden High School              83.803279             92.505855   
Shelton High School             83.359455             93.867121   
Wright High School              83.682222             93.333333   
Wilson High School              83.274201             93.867718   
Griffin High School             83.351499             93.392371   
Pena High School                83.839917             94.594595   
Thomas High School              83.418349             93.272171   
Cabrera High School             83.061895             94.133477   

                       Passing Reading Percent  Passing Rate  
school_name                                                   
Rodriguez High School                80.220055     73.293323  
Figueroa High School                 80.739234     73.363852  
Huang High School                    81.316421     73.500171  
Johnson High School                  81.222432     73.639992  
Ford High School                     79.299014     73.804308  
Hernandez High School                80.862999     73.80798

In [282]:
# Save the results to a dataframe called top_schools
top_schools = per_school_summary.sort_values("Passing Rate", ascending=True).head().copy()


# Bottom Performing Schools (By % Overall Passing)

In [283]:
#Sort lowest perform schools by Percentage of Overall Passing
bottom_perform = per_school_summary.sort_values("Passing Rate", ascending=False).tail()
bottom_perform.style.format({
    'Total Students': '{:}',
    'Total School Budget': "${:,.2f}", 
    'Per Student Budget': "${:.2f}", 
    'Passing Math Percent': "{:6f}", 
    'Passing Reading Percent': "{:6f}", 
    'Passing Rate': "{:6f}"
})


bottom_perform


School Type  Total Students Total School Budget  \
school_name                                                             
Ford High School         District            2739       $1,763,916.00   
Johnson High School      District            4761       $3,094,650.00   
Huang High School        District            2917       $1,910,635.00   
Figueroa High School     District            2949       $1,884,411.00   
Rodriguez High School    District            3999       $2,547,363.00   

                      Per Student Budget  Average Reading Score  \
school_name                                                       
Ford High School                 $644.00              80.746258   
Johnson High School              $650.00              80.966394   
Huang High School                $655.00              81.182722   
Figueroa High School             $639.00              81.158020   
Rodriguez High School            $637.00              80.744686   

                       Average_Math_Score  Passing Math Percent  \
school_name                                                       
Ford High School                77.102592             68.309602   
Johnson High School             77.072464             66.057551   
Huang High School               76.629414             65.683922   
Figueroa High School            76.711767             65.988471   
Rodriguez High School           76.842711             66.366592   

                       Passing Reading Percent  Passing Rate  
school_name                                                   
Ford High School                     79.299014     73.804308  
Johnson High School                  81.222432     73.639992  
Huang High School                    81.316421     73.500171  
Figueroa High School                 80.739234     73.363852  
Rodriguez High School                80.220055     73.293323

In [284]:
# Save the results to a dataframe called bottom_schools
bottom_schools = per_school_summary.sort_values("Passing Rate", ascending=False).tail().copy()

# Math Scores by Grade

In [285]:
# Use the code provided to separate the data by grade, group by school_name, and take the mean of each
ninth_math = student_data.loc[student_data['grade'] == '9th'].groupby('school_name')["math_score"].mean()
tenth_math = student_data.loc[student_data['grade'] == '10th'].groupby('school_name')["math_score"].mean()
eleventh_math = student_data.loc[student_data['grade'] == '11th'].groupby('school_name')["math_score"].mean()
twelfth_math = student_data.loc[student_data['grade'] == '12th'].groupby('school_name')["math_score"].mean()

# Combine each of the scores above into single DataFrame called math_scores_by_grade (4 points)

math_scores_by_grade = pd.DataFrame({
        "9th": ninth_math,
        "10th": tenth_math,
        "11th": eleventh_math,
        "12th": twelfth_math
})
math_scores = math_scores_by_grade[['9th', '10th', '11th', '12th']]
math_scores.index.name = "School Name"

#show and format
math_scores_by_grade.style.format({'9th': '{:.6f}', 
                          "10th": '{:.6f}', 
                          "11th": "{:.6f}", 
                          "12th": "{:.6f}"})

# Reading Scores by Grade 


In [286]:
#Use the code provided to separate the data by grade, group by school_name, and take the mean of each
ninth_reading = student_data.loc[student_data['grade'] == '9th'].groupby('school_name')["reading_score"].mean()
tenth_reading = student_data.loc[student_data['grade'] == '10th'].groupby('school_name')["reading_score"].mean()
eleventh_reading = student_data.loc[student_data['grade'] == '11th'].groupby('school_name')["reading_score"].mean()
twelfth_reading = student_data.loc[student_data['grade'] == '12th'].groupby('school_name')["reading_score"].mean()


#Combine each of the scores above into single DataFrame called reading_scores_by_grade
reading_scores_by_grade = pd.DataFrame({
        "9th": ninth_reading,
        "10th": tenth_reading,
        "11th": eleventh_reading,
        "12th": twelfth_reading
})
reading_scores = reading_scores_by_grade[['9th', '10th', '11th', '12th']]
reading_scores.index.name = "School Name"

#format
reading_scores_by_grade.style.format({'9th': '{:.6f}', 
                             "10th": '{:.6f}', 
                             "11th": "{:.6f}", 
                             "12th": "{:.6f}"})

# Scores by School Spending



In [287]:
#Use pd.cut with the provided code to bin the data by the spending ranges (2 points)
#Use the code provided to calculate the averages (1 points)
#Create the spending_summary DataFrame using the binned and averaged spending data (2 points)

bins = [0, 584, 629, 644, 675]
group_name = ["<$584", "$585-629", "$630-644", "$645-675"]

school_data_complete['spending_bins'] = pd.cut(school_data_complete['budget']/school_data_complete['size'], bins, labels = group_name)

#group by spending
by_spending = school_data_complete.groupby('spending_bins')


avg_math = by_spending['math_score'].mean()

avg_read = by_spending['reading_score'].mean()
pass_math = school_data_complete[school_data_complete['math_score'] >= 70].groupby('spending_bins')['Student ID'].count()/by_spending['Student ID'].count()*100
pass_read = school_data_complete[school_data_complete['reading_score'] >= 70].groupby('spending_bins')['Student ID'].count()/by_spending['Student ID'].count()*100
overall = school_data_complete[(school_data_complete['reading_score'] >= 70) & (school_data_complete['math_score'] >= 70)].groupby('spending_bins')['Student ID'].count()/by_spending['Student ID'].count()*100
    
# df build            
scores_by_spend = pd.DataFrame({
    "Average Math Score": avg_math,
    "Average Reading Score": avg_read,
    "% Passing Math": pass_math,
    "% Passing Reading": pass_read,
    "% Overall Passing": overall
            
})
            
#reorder columns
scores_by_spend = scores_by_spend[[
    "Average Math Score",
    "Average Reading Score",
    "% Passing Math",
    "% Passing Reading",
    "% Overall Passing"
]]

scores_by_spend.index.name = "Per Student Budget"


#formating
scores_by_spend_formatted = scores_by_spend.style.format({'Average Math Score': '{:.2f}', 
                                                          'Average Reading Score': '{:.2f}', 
                                                          '% Passing Math': '{:.2f}', 
                                                          '% Passing Reading':'{:.2f}', 
                                                          '% Overall Passing': '{:.2f}'})
scores_by_spend_formatted

# Scores by School Size 

In [288]:
# create size bins
bins = [0, 1000, 1999,5000]
group_name = ["Small (<1000)", "Medium (1000-2000)" , "Large (2000-5000)"]
school_data_complete['size_bins'] = pd.cut(school_data_complete['size'], bins, labels = group_name)

#group by spending
by_size = school_data_complete.groupby('size_bins')

#calculations 
average_math_score = by_size['math_score'].mean()
average_reading_score = by_size['math_score'].mean()
pass_math_percent = school_data_complete[school_data_complete['math_score'] >= 70].groupby('size_bins')['Student ID'].count()/by_size['Student ID'].count()*100
pass_read_percent = school_data_complete[school_data_complete['reading_score'] >= 70].groupby('size_bins')['Student ID'].count()/by_size['Student ID'].count()*100
overall = school_data_complete[(school_data_complete['reading_score'] >= 70) & (school_data_complete['math_score'] >= 70)].groupby('size_bins')['Student ID'].count()/by_size['Student ID'].count()*100

            
# df build            
scores_by_size = pd.DataFrame({
    "Average Math Score": average_math_score,
    "Average Reading Score": average_reading_score,
    '% Passing Math': pass_math_percent,
    '% Passing Reading': pass_read_percent,
    '% Overall Passing': overall
            
})
            
#reorder columns
scores_by_size = scores_by_size[[
    "Average Math Score",
    "Average Reading Score",
    '% Passing Math',
    '% Passing Reading',
    '% Overall Passing'
]]

scores_by_size.index.name = "Total Students"
scores_by_size = scores_by_size.reindex(group_name)

#formating
scores_by_size.style.format({'Average Math Score': '{:.6f}', 
                              'Average Reading Score': '{:.6f}', 
                              '% Passing Math': '{:.6f}', 
                              '% Passing Reading':'{:.6f}', 
                              '% Overall Passing': '{:.6f}'})

# Scores by School Type

In [289]:
# group by type of school
school_type = school_data_complete.groupby("type")

#calculations 
average_math_score = school_type['math_score'].mean()
average_reading_score = school_type['math_score'].mean()
pass_math_percent = school_data_complete[school_data_complete['math_score'] >= 70].groupby('type')['Student ID'].count()/school_type['Student ID'].count()*100
pass_read_percent = school_data_complete[school_data_complete['reading_score'] >= 70].groupby('type')['Student ID'].count()/school_type['Student ID'].count()*100
overall = school_data_complete[(school_data_complete['reading_score'] >= 70) & (school_data_complete['math_score'] >= 70)].groupby('type')['Student ID'].count()/school_type['Student ID'].count()*100

# df build            
scores_school_type = pd.DataFrame({
    "Average Math Score": average_math_score,
    "Average Reading Score": average_reading_score,
    '% Passing Math': pass_math_percent,
    '% Passing Reading': pass_read_percent,
    "% Overall Passing": overall})
    
#reorder columns
scores_school_type = scores_schoo_type[[
    "Average Math Score",
    "Average Reading Score",
    '% Passing Math',
    '% Passing Reading',
    "% Overall Passing"
]]
scores_school_type.index.name = "Type of School"


#formating
scores_school_type.style.format({'Average Math Score': '{:.6f}', 
                              'Average Reading Score': '{:.6f}', 
                              '% Passing Math': '{:.6f}', 
                              '% Passing Reading':'{:.6f}', 
                              '% Overall Passing': '{:.6f}'})

# Written Report

#Summarizes the analysis
The analysis reveals key insights into the district and individual schools. The district, comprised of 15 schools and 39,170 students, shows an average math score of 78.99 and reading score of 81.88, with an overall passing rate of 65.17%. Charter schools consistently outperform district schools, with top-performing schools including Cabrera, Thomas, Griffin, Wilson, and Pena. On the other hand, bottom-performing schools like Rodriguez, Figueroa, Huang, Hernandez, and Johnson exhibit lower passing rates. 


#Draws two correct conclusions or comparisons from the calculations
From the analysis, we can assume that charter schools outperform district schools in both average math and reading scores as well as overall passing rates. This couldpossibly mean that charter schools have higher academic success as compared to the district schools. 

The sizing of school also plays a role in academic success. For example, smaller-sized schools had higher averaged math and reading scores, as well as better overall passing rates. This is in comparison to larger schools. This is helpful for educational policymakers who are working on how to strategize to improve academic success in schools. 